docs/src/test-sharding-js.md

```diff
@@ -0,0 +1,169 @@
+
+### Configuring shard reporting
+
+We start by adding `blob` reporter to the config:
+
+```js
+export default defineConfig({
+  testDir: './tests',
+  reporter: [['blob', { outputDir: 'blob-report' }]],
+});
+```
+
+Blob report contains information about all the tests that were run and their results as well as all test attachments such as traces and screenshot diffs. Blob reports can be merged and converted to any other Playwright report.
+
+### Uploading shard report
+
+To merge individual reports they need to be copied into a shared location. GitHub Actions Artifacts is a convenient mechanism that lets you do that. By adding the following step after the test execution we upload blob report from each shard into GitHub Actions Artifact with name`blob-report-${{ github.run_attempt }}` (this is essentially a shared directory where each shard will copy its report to):
+
+```yaml
+jobs:
+  test:
+...
+    - name: Upload blob report to Artifacts
+      if: always()
+      uses: actions/upload-artifact@v3
+      with:
+        name: blob-report-${{ github.run_attempt }}
+        path: blob-report
+        retention-days: 2
+```
+
+### Merge reports job
+
+After all shards finished running it's time to run a job that will merge the reports and produce a combined HTML report. To ensure the execution order, we make `merge-report` job [depend](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#defining-prerequisite-jobs) on our sharded `test` job. `merge-report` job below reads all blob reports from `blob-report-${{ github.run_attempt }}` artifact and produces a single HTML report.
+
+```yaml
+jobs:
+  merge-report:
+    if: always()
+    needs: [test]
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 18
+    - name: Install dependencies
+      run: npm ci
+
+    - name: Download Blob Reports from Artifacts
+      uses: actions/download-artifact@v3
+      with:
+        name: blob-report-${{ github.run_attempt }}
+        path: blob-report
+
+    - name: Merge into HTML Report
+      run: |
+        npx playwright merge-reports ./blob-report --reporter html
+
+    - name: Upload HTML report
+      uses: actions/upload-artifact@v3
+      with:
+        name: html-report-${{ github.run_attempt }}
+        path: playwright-report
+```
+
+`merge-report` job will run even if there were test failures and it wil write the HTML report into `playwright-report` directory by default.
+
+### Serving report: GitHub Actions Artifacts 
+
+Merged HTML report can be saved as `html-report-${{ github.run_attempt }}` artifact by adding following step:
+
+```yaml
+    - name: Upload HTML report
+      uses: actions/upload-artifact@v3
+      with:
+        name: html-report-${{ github.run_attempt }}
+        path: playwright-report
+```
+
+You can download the artifact via GitHub UI to see the report. This method has its own pros and cons:
+
+Pros:
```

Comment:
Since other sections below do not have explicit Pros/Cons, I'd rewrite them here as well:


```
In the example above, we upload the html report as GitHub Actions Artifact. This is easy to configure, but downloading html report as a zip file is not very convenient.

In the next section, we'll illustrate how to make the report accessible from cloud storage.
```